tests/validator/validator.cpp

```diff
@@ -852,143 +857,118 @@ TEST(Validator, integerStrings)
+        "Reset in component 'comp' with order '6', with variable 'var', with test_variable 'var2', does not have a reset_value specified.",
+        "Reset in component 'comp' with order '7', with variable 'var', with test_variable 'var2', does not have a test_value specified.",
     };
 
     libcellml::ModelPtr m = std::make_shared<libcellml::Model>();
     libcellml::ComponentPtr c = std::make_shared<libcellml::Component>();
-    libcellml::VariablePtr var = std::make_shared<libcellml::Variable>();
+    libcellml::VariablePtr v1 = std::make_shared<libcellml::Variable>();
+    libcellml::VariablePtr v2 = std::make_shared<libcellml::Variable>();
     libcellml::ResetPtr r1 = std::make_shared<libcellml::Reset>();
     libcellml::ResetPtr r2 = std::make_shared<libcellml::Reset>();
     libcellml::ResetPtr r3 = std::make_shared<libcellml::Reset>();
     libcellml::ResetPtr r4 = std::make_shared<libcellml::Reset>();
     libcellml::ResetPtr r5 = std::make_shared<libcellml::Reset>();
     libcellml::ResetPtr r6 = std::make_shared<libcellml::Reset>();
     libcellml::ResetPtr r7 = std::make_shared<libcellml::Reset>();
-    libcellml::WhenPtr w1 = std::make_shared<libcellml::When>();
-    libcellml::WhenPtr w2 = std::make_shared<libcellml::When>();
-
-    w1->setOrder(776);
-    w1->setCondition(emptyMath);
-    w1->setValue(emptyMath);
-    w2->setOrder(345);
-    w2->setCondition(emptyMath);
-    w2->setValue(emptyMath);
-
-    r1->setOrder(300);
-    r1->addWhen(w1);
-    r6->addWhen(w1);
-    r2->setOrder(300);
-    r2->addWhen(w1);
-    r2->addWhen(w2);
-    r2->setVariable(var);
-    r3->setOrder(400);
-    r3->addWhen(w2);
-    r3->setVariable(var);
-    r4->setVariable(var);
-    r4->setOrder(500);
-    r5->setVariable(var);
+
+    // This one is good :)
+    r1->setVariable(v1);
+    r1->setTestVariable(v2);
+    r1->setOrder(1);
+    r1->setResetValue(goodMath);
+    r1->setTestValue(goodMath);
+
+    // Duplicated order in the same variable
+    r2->setVariable(v1);
+    r2->setTestVariable(v2);
+    r2->setOrder(1);
+    r2->setResetValue(goodMath);
+    r2->setTestValue(goodMath);
+
+    // No variable
+    //r3->setVariable(v1);
+    r3->setTestVariable(v2);
+    r3->setOrder(3);
+    r3->setResetValue(goodMath);
+    r3->setTestValue(goodMath);
+
+    // No test_variable
+    r4->setVariable(v1);
+    //r4->setTestVariable(v2);
+    r4->setOrder(4);
+    r4->setResetValue(goodMath);
+    r4->setTestValue(goodMath);
+
+    // No order
+    r5->setVariable(v1);
+    r5->setTestVariable(v2);
+    // r5->setOrder(5);
+    r5->setResetValue(goodMath);
+    r5->setTestValue(goodMath);
+
+    // No reset_value
+    r6->setVariable(v1);
+    r6->setTestVariable(v2);
+    r6->setOrder(6);
+    // r6->setResetValue(goodMath);
+    r6->setTestValue(goodMath);
+
+    // No test_value
+    r7->setVariable(v1);
+    r7->setTestVariable(v2);
+    r7->setOrder(7);
+    r7->setResetValue(goodMath);
+    // r7->setTestValue(emptyMath);
```

Comment:
would be good to test both options of either not setting the math or setting with an empty string.

Reply:
I've added a test for whitespace only, which returns an equivalent error to reset-value-not-set etc ... but am confused as to whether empty math blocks should be valid or not?  See issue #365

Reply:
Done, including empty maths as per discussion in #365.